src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpoint.java

```
@@ -236,6 +236,9 @@ public String imageName(String userAndRepo) throws IOException {
         if (effectiveUrl.getPort() > 0 && effectiveUrl.getDefaultPort() != effectiveUrl.getPort()) {
             s.append(':').append(effectiveUrl.getPort());
         }
+        if (userAndRepo.startsWith(String.valueOf(s))) {
```

Agreed, better to propagate the error